test/python/circuit/test_circuit_load_from_qpy.py

```diff
@@ -865,3 +865,14 @@ def test_standalone_register_partial_bit_in_circuit(self):
         qpy_file.seek(0)
         new_circuit = load(qpy_file)[0]
         self.assertEqual(qc, new_circuit)
+
+    def test_nested_tuple_param(self):
+        """Test qpy with an instruction that contains nested tuples."""
+        inst = Instruction("tuple_test", 1, 0, [((((0, 1), (0, 1)), 2, 3), ("A", "B", "C"))])
+        qc = QuantumCircuit(1)
+        qc.append(inst, [0])
+        qpy_file = io.BytesIO()
+        dump(qc, qpy_file)
+        qpy_file.seek(0)
+        new_circuit = load(qpy_file)[0]
+        self.assertEqual(qc, new_circuit)
```

Comment:
This looks like as much nesting as we need to bother with.  A test of the empty tuple somewhere in there probably wouldn't hurt as an edge case, but that's it, and I'm fairly confident it should already work.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7584/commits/e417e7549962f97d12a1aa2fdccde304242e8562